pandas/core/dtypes/common.py

```diff
@@ -1787,92 +1762,67 @@ def _get_dtype(arr_or_dtype):
     TypeError : The passed in object is None.
     """
 
-    # TODO(extension)
-    # replace with pandas_dtype
-
     if arr_or_dtype is None:
         raise TypeError("Cannot deduce dtype from null object")
-    if isinstance(arr_or_dtype, np.dtype):
+
+    # fastpath
+    elif isinstance(arr_or_dtype, np.dtype):
```

Comment:
you caught ExtensionDType etc in fastpath below.  is there a reason not to do it here?